implement ligthweight tags

christianguenter2 opened the pull request:
First basic support for ligthweight tags. #1081 

~~The feature can be enabled with the 'Enable experimental features' switch.~~

It works quite similar to branches, although you cannot checkout tags yet.
![image](https://user-images.githubusercontent.com/17437789/34487712-d77b35aa-efd5-11e7-84d0-98e508325699.png)

Overview
![image](https://user-images.githubusercontent.com/17437789/34501165-060ccedc-f00e-11e7-8e77-1f00e41341b4.png)

˙
Create tag
![image](https://user-images.githubusercontent.com/17437789/34487846-560afd9c-efd6-11e7-8a16-6f7649f71367.png)
![image](https://user-images.githubusercontent.com/17437789/34487851-58019fa2-efd6-11e7-80b0-265a8e9fe03f.png)


Delete tag
![image](https://user-images.githubusercontent.com/17437789/34487874-7bf46052-efd6-11e7-9aa9-8184586da3e4.png)
![image](https://user-images.githubusercontent.com/17437789/34487890-87b5faf4-efd6-11e7-88c2-c856781ad1c2.png)

larshp commented:
Looks good, few comments:

A: Can you add a comment in create_tag that it is a "lightweight" tag?

B: Somehow we should make it visible for the user that the tag is created from the current commit

C: I'm okay with not adding this as an experimental feature, it is a new feature, so it should not affect the current usage of abapGot

christianguenter2 commented:
Good points;

A: I added an explaining comment 
B: I added a popup showing the SHA of the current commit
![image](https://user-images.githubusercontent.com/17437789/34500622-d895661e-f00b-11e7-910c-61e3622d6f48.png)

C: I removed the activation with the experimental feature flag

larshp commented:
Looks good to me, tested successfully, have a few comments which I will add in the isssue, lets get this merged 🚢
